src/common/reducers/repositories.js

```diff
@@ -2,10 +2,11 @@ import * as ActionTypes from '../actions/repositories';
 import { parseGitHubRepoUrl } from '../helpers/github-url';
 
 export function repositories(state = {
-  isFetching: false,
+  // has no server side content, so initial state is always 'fetching'
+  isFetching: true,
```

Comment:
It seems to make sense, but it also seems to be making things more complicated.

I agree that initial empty state needs we are in `needData: true` state. But it doesn't mean we are in `isFetching: true` state (which means we already triggered a request and are waiting for response).

While it may not be the case in this particular reducer we use the same `isFetching` state everywhere else and in some places there are explicit checks for `isFetching` being true to make sure we don't try to fetch data again while we are waiting for response.
By setting `isFetching` to true in initial state we would break it and data will never load (because components that request the data will assume we are already waiting for it).

I understand the reasoning behind it. Right now our server side rendering doesn't do much - it renders incomplete empty initial state, without spinner even (because actions have not triggered yet), but I don't think using `isFetching` is a good place to fix that.

On the other hand - what is really the benefit of server side rendering the spinner?